homeassistant/components/yamaha_musiccast/media_player.py

```diff
@@ -408,3 +431,364 @@ def media_position_updated_at(self):
             return self.coordinator.data.netusb_play_time_updated
 
         return None
+
+    # Group and MusicCast System specific functions/properties
+
+    @property
+    def is_network_server(self) -> bool:
+        """Return only true if the current entity is a network server and not a main zone with an attached zone2."""
+        return (
+            self.coordinator.data.group_role == "server"
+            and self.coordinator.data.group_id != NULL_GROUP
+            and self._zone_id == self.coordinator.data.group_server_zone
+        )
+
+    @property
+    def other_zones(self) -> list[MusicCastMediaPlayer]:
+        """Return media player entities of the other zones of this device."""
+        return [
+            entity
+            for entity in self.coordinator.entities
+            if entity != self and isinstance(entity, MusicCastMediaPlayer)
```

Comment:
All of the entities in this platform are `MusicCastMediaPlayer` instances. Why do we have the check for that?

If we are worried about mixing them with other platform entities: Entities are a platform concern. It's usually a bad pattern to store them centrally.